src/components/settings/SettingsLMSTab/index.jsx

```diff
@@ -137,9 +137,11 @@ const SettingsLMSTab = ({
 
   useEffect(() => {
     // update list of used display names to prevent duplicates
+    const updatedMap = new Map();
     if (existingConfigsData[0]) {
-      setDisplayNames(existingConfigsData?.map((existingConfig) => existingConfig.displayName));
+      existingConfigsData?.map((existingConfig) => updatedMap.set(existingConfig.displayName, existingConfig.id));
```

Comment:
No reason to use .map if we're not using the functionality of .map to generate the Map.
```suggestion
      existingConfigsData?.forEach((existingConfig) => updatedMap.set(existingConfig.displayName, existingConfig.id));
```